heron/scheduler-core/src/java/com/twitter/heron/scheduler/SubmitterMain.java

```diff
@@ -365,22 +381,30 @@ public boolean submitTopology() {
     String uploaderClass = Context.uploaderClass(config);
     IUploader uploader;
 
+    // create an instance of state manager
     try {
-      // create an instance of state manager
       statemgr = ReflectionUtils.newInstance(statemgrClass);
+    } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
+      throw new TopologySubmissionException(
+          String.format("Failed to instantiate state manager class '%s'", statemgrClass), e);
+    }
 
-      // create an instance of launcher
+    // create an instance of launcher
+    try {
       launcher = ReflectionUtils.newInstance(launcherClass);
+    } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
+      throw new TopologySubmissionException(
+          String.format("Failed to instantiate launcher class '%s'", launcherClass), e);
+    }
 
-      // create an instance of uploader
+    // create an instance of uploader
+    try {
       uploader = ReflectionUtils.newInstance(uploaderClass);
     } catch (IllegalAccessException | InstantiationException | ClassNotFoundException e) {
-      LOG.log(Level.SEVERE, "Failed to instantiate instances", e);
-      return false;
+      throw new TopologySubmissionException(
```

Comment:
how about instantiation of state manager and launcher class? should we also create exception for them?